exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py

```diff
@@ -0,0 +1,199 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Sequence
+
+from opentelemetry.exporter.prometheus_remote_write.gen.types_pb2 import (
+    Label,
+    Sample,
+    TimeSeries,
+)
+from opentelemetry.sdk.metrics.export import (
+    ExportRecord,
+    MetricsExporter,
+    MetricsExportResult,
+)
+
+
+class PrometheusRemoteWriteMetricsExporter(MetricsExporter):
+    """
+    Prometheus remote write metric exporter for OpenTelemetry.
+
+    Args:
+        endpoint: url where data will be sent (Required)
+        basic_auth: username and password for authentication (Optional)
+        headers: additional headers for remote write request (Optional)
+        timeout: timeout for requests to the remote write endpoint in seconds (Optional)
+        proxies: dict mapping request proxy protocols to proxy urls (Optional)
+        tls_config: configuration for remote write TLS settings (Optional)
+    """
+
+    def __init__(
+        self,
+        endpoint: str,
+        basic_auth: Dict = None,
+        headers: Dict = None,
+        timeout: int = 30,
+        tls_config: Dict = None,
+        proxies: Dict = None,
+    ):
+        self.endpoint = endpoint
+        self.basic_auth = basic_auth
+        self.headers = headers
+        self.timeout = timeout
+        self.tls_config = tls_config
+        self.proxies = proxies
+
+    @property
+    def endpoint(self):
+        return self._endpoint
+
+    @endpoint.setter
+    def endpoint(self, endpoint: str):
+        if endpoint == "":
+            raise ValueError("endpoint required")
+        self._endpoint = endpoint
+
+    @property
+    def basic_auth(self):
+        return self._basic_auth
+
+    @basic_auth.setter
+    def basic_auth(self, basic_auth: Dict):
+        if basic_auth:
+            if "username" not in basic_auth:
+                raise ValueError("username required in basic_auth")
+            if (
+                "password" not in basic_auth
+                and "password_file" not in basic_auth
+            ):
+                raise ValueError("password required in basic_auth")
+            if "password" in basic_auth and "password_file" in basic_auth:
+                raise ValueError(
+                    "basic_auth cannot contain password and password_file"
+                )
+        self._basic_auth = basic_auth
+
+    @property
+    def timeout(self):
+        return self._timeout
+
+    @timeout.setter
+    def timeout(self, timeout: int):
+        if timeout <= 0:
+            raise ValueError("timeout must be greater than 0")
+        self._timeout = timeout
+
+    @property
+    def tls_config(self):
+        return self._tls_config
+
+    @tls_config.setter
+    def tls_config(self, tls_config: Dict):
+        if tls_config:
+            new_config = {}
```

Comment:
This does not seem to be used.

Reply:
yup, this will be used in PR 4/6 when we actually make the POST request

Reply:
As mentioned by Azfaar, it's being assigned in the contructor: `self.tls_config = tls_config` and will be used in a later PR when making the request.